app/src/main/java/com/mpieterse/stride/core/services/NotificationSchedulerService.kt

```
@@ -0,0 +1,294 @@
+            val soundEnabled = inputData.getBoolean("sound_enabled", true)
+            val vibrationEnabled = inputData.getBoolean("vibration_enabled", true)
+            
+            Clogger.d("NotificationWorker", "Executing notification work for: $habitName")
+            Clogger.d("NotificationWorker", "Notification ID: $notificationId, Title: $title, Body: $body")
+            
+            // Display notification
+            displayNotification(notificationId, title, body, soundEnabled, vibrationEnabled)
+            
+            Clogger.d("NotificationWorker", "Notification work completed successfully")
+            Result.success()
+        } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is too generic. Prefer catching specific exceptions to the case that is currently handled.

(detekt.exceptions.TooGenericExceptionCaught)